client/components/average-grade.jsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+function AverageGrade(props) {
+  let totalSumOfGrades = 0;
+  let averageGrade = 0;
+  const grades = props.average;
+  for (let i = 0; i < grades.length; i++) {
+    totalSumOfGrades += grades[i].grade;
+  }
+  if (grades.length) {
+    averageGrade = Math.floor(totalSumOfGrades / grades.length);
+  }
```

Comment:
Instead of passing all of the grades to the average, calculate the average in App and send the average grade down so that AverageGrade doesn't have extra chores to do.